main/src/cgeo/geocaching/ui/CalculatorVariable.java

```diff
@@ -0,0 +1,283 @@
+package cgeo.geocaching.ui;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Color;
+import android.graphics.Typeface;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.GridLayout;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
+import android.util.TypedValue;
+import android.view.Gravity;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.io.Serializable;
+import java.util.List;
+
+import cgeo.geocaching.R;
+
+/**
+ * Class used to display a variable with an equation, such as:
+ *  X = a^2 = b^2
+ */
+
+public class CalculatorVariable extends LinearLayout {
+
+    // Data used to capture the state of this Variable such that it can be restored again later.
+    public static class VariableData implements Serializable {
+        char name;
+        String expression; // Note, we have to use a String rather than an Editable as Editable's can't be serialized
+
+        public VariableData(char theName) {
+            name = theName;
+            expression = "";
+        }
+    }
+
+    private VariableData variableData;
+
+    // Views used to display the name and expression of the variable.
+    // Note that the actual name and expression used are stored in the variableData 'struct'
+    // and these views are only used to display that information
+    private TextView name;
+    private EditText expression;
+
+    // Variables used to save unnecessary re-computing.
+    private double cachedValue;
+    private boolean cacheDirty;
+
+    public CalculatorVariable(Context context, VariableData theData, String hintText, TextWatcher textWatcher) {
```

Comment:
Codacy forgot to flag the missing `final` here (also, as everywhere, please remove the `the` prefix to get your code in the same style as the rest of cgeo).